lib/private/Files/SimpleFS/NewSimpleFile.php

```diff
@@ -190,6 +190,17 @@ public function getMimeType() {
 		}
 	}
 
+	/**
+	 * {@inheritDoc}
+	 */
+	public function getExtension(): string {
+		if ($this->file) {
+			return $this->file->getExtension();
+		} else {
+			return 'txt';
```

Comment:
I think this should work:
```suggestion
			return \pathinfo($this->name, PATHINFO_EXTENSION);
```

Reply:
Using this for InMemoryFile as well since `new \finfo(FILEINFO_EXTENSION)` doesn't work